components/core/src/clp/CurlGlobalInstance.cpp

```diff
@@ -0,0 +1,45 @@
+#include "CurlGlobalInstance.hpp"
+
+#include <mutex>
+
+#include <curl/curl.h>
+
+#include "CurlOperationFailed.hpp"
+#include "ErrorCode.hpp"
+
+namespace clp {
+CurlGlobalInstance::CurlGlobalInstance() {
+    std::scoped_lock<std::mutex> const global_lock{m_ref_count_mutex};
```

Comment:
```suggestion
    std::scoped_lock const global_lock{m_ref_count_mutex};
```

One of the benefits of `scoped_lock` is that it supports template argument deduction.